src/twisted/conch/ssh/filetransfer.py

```diff
@@ -856,12 +856,8 @@ def _cbReadDir(self, names):
 
 
     def _ebReadDir(self, reason):
-        reason.trap(EOFError)
-        def _():
-            raise StopIteration
-        self.__next__ = _
-        self.next = self.__next__
```

Comment:
This change removes the monkeypatching of `next`/`__next__` (which I think is good, stylistically) but it changes the behavior of the iterator. It no longer remembers that it has reached EOF, and calling `next` again will cause a protocol violation (issuing `READDIR` on a closed handle). Python iterators [are required](https://docs.python.org/3/library/stdtypes.html#iterator.__next__) to allow calling `next` after StopIteration (and should continue raising StopIteration).